README.md

```diff
@@ -1,39 +1,431 @@
+[![Code Climate](https://codeclimate.com/github/athityakumar/daru-io.png)](https://codeclimate.com/github/athityakumar/daru-io)
+[![Stories in Ready](https://badge.waffle.io/athityakumar/daru-io.png?label=ready&title=Ready)](https://waffle.io/athityakumar/daru-io?utm_source=badge)
+[![License: MIT](https://img.shields.io/badge/License-MIT-blue.svg)](https://opensource.org/licenses/MIT)
 
-TODO: Delete this and the text above, and describe your gem
+A Ruby plugin-gem to [daru gem](https://github.com/SciRuby/daru), that extends support for many Import and
+Export methods of `Daru::DataFrame`. This gem is intended to help Data analyzing and Web Developer Rubyists,
+by serving as a general purpose conversion library that takes input in one format (say, JSON) and
+converts it another format (say, Avro) while making it incredibly easy to analyze the data.
 
-## Installation
+# Table of contents
 
-Add this line to your application's Gemfile:
+- [Installation](#installation)
+- *[Importers](#importers): [ActiveRecord](#activerecord-importer), [Avro](#avro-importer), [CSV](#csv-importer), [Excel](#excel-importer), [Excelx](#excelx-importer), [HTML](#html-importer), [JSON](#json-importer), [Mongo](#mongo-importer), [Plaintext](#plaintext-importer), [RData](#rdata-importer), [RDS](#rds-importer), [Redis](#redis-importer), [SQL](#sql-importer)*
+- *[Exporters](#exporters): [Avro](#avro-exporter), [CSV](#csv-exporter), [Excel](#excel-exporter), [JSON](#json-exporter), [RData](#rdata-exporter), [RDS](#rds-exporter), [SQL](#sql-exporter)*
+- [Contributing](#contributing)
+- [License](#license)
+
+# Installation
+
+[(Go to Table of Contents)](#table-of-contents)
+
+- If you're working with a Gemfile,
+
+    - Add this line to your application's Gemfile:
+
+    ```ruby
+    gem 'daru-io'
+    ```
+
+    - And then execute:
+
+    ```sh
+    bundle
+    ```
+
+- If you're NOT working with a Gemfile, simply install it yourself as:
+
+  ```sh
+  gem install daru-io
+  ```
+
+- Require `daru-io` gem in your application:
+
+```ruby
+require 'daru/io' #! Requires all Importers & Exporters
+require 'daru/io/importers' #! Requires all Importers and not Exporters
+require 'daru/io/importers/json' #! Requires only JSON Importer
+```
+
+*Note - Each IO module has it's own set of dependencies. Have a look at the [Importers](#importers) and [Exporters](#exporters) section for dependency-specific information.*
+
+# Importers
+
+### ActiveRecord Importer
+
+[(Go to Table of Contents)](#table-of-contents)
+
+```ruby
+#! Partially require just ActiveRecord Importer
+require 'daru/io/importers/active_record'
+
+#! Usage from Daru::IO
+Daru::IO::Importers::ActiveRecord.new(args).call
+
+#! Usage from Daru::DataFrame
+Daru::DataFrame.from_activerecord(args)
+```
+
+- Have a look at [this documentation](http://www.rubydoc.info/github/athityakumar/daru-io/master/Daru/IO/Importers/ActiveRecord) to know more about this Importer.
+- Dependency : activerecord gem.
```

Comment:
Maybe instead of repeating "Have a look...." do something short and formal, like this:

### ActiveRecord

* **Dependencies**: `activerecord`
* **Usage**:
```ruby
require 'daru/io/importers/active_record'
Daru::DataFrame.from_activerecord(connection) # and maybe some useful options example
```
* **Docs**: [RubyDoc.info](http://www.rubydoc.info/github/athityakumar/daru-io/master/Daru/IO/Importers/ActiveRecord)


WDYT?